MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift

```diff
@@ -25,7 +24,13 @@ class MapboxCoreNavigationTests: XCTestCase {
     func testDepart() {
         route.accessToken = "foo"
         navigation = MapboxNavigationService(route: route, directions: directions, simulating: .never)
-        let depart = CLLocation(coordinate: CLLocationCoordinate2D(latitude: 37.795042, longitude: -122.413165), altitude: 1, horizontalAccuracy: 1, verticalAccuracy: 1, course: 0, speed: 10, timestamp: Date())
+        
+        // Coordinates from first step
+        let coordinates = route.legs[0].steps[0].coordinates!
+        let locations = coordinates.enumerated().map { CLLocation(coordinate: $0.element,
+                                                                  altitude: -1, horizontalAccuracy: 10,
+                                                                  verticalAccuracy: -1, course: -1, speed: 10,
+                                                                  timestamp: Date().addingTimeInterval(TimeInterval($0.offset))) }
```

Comment:
I’m assuming these offsets are all measured from the same starting time. In general, to ensure accuracy, a single Date instance should be created outside the call to `map(_:)`, and this line would add a time interval to that single Date instance.

Also, it looks like this PR adds a `+` operator for Date and TimeInterval that could be used here.

(×5)